File: Gosuji/Gosuji.Client/Models/KataGo/Move.cs
using System.ComponentModel.DataAnnotations;

namespace Gosuji.Client.Models.KataGo
{
    public class Move
    {
        [Required]
        [RegularExpression(@"(B|W)")]
        public string color { get; set; }
        [Required]
        [RegularExpression(@"([A-H]|[J-T])(1[0-9]|[1-9])")]
        public string coord { get; set; }

        public Move() { }

        public Move(string color, string coord)
        {
            this.color = color;
            this.coord = coord;
        }

        public override string ToString()
        {
            return "{ color: " + color +
                ", coord: " + coord + " }";
        }
    }
}

File: Gosuji/Gosuji.API/Resources/js/classes/Move.js
let Move = window.Move;

if (typeof Move === "undefined") {
    Move = class Move {
        color;
        coord;
    
        constructor(color, coord) {
            this.color = color;
            this.coord = coord;
        }
    }

    window.Move = Move;
}

export { Move };
